python/tvm/relax/training/trainer.py

```diff
@@ -0,0 +1,253 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name
+"""Unified Trainer API for relax training."""
+
+from typing import Union, List, Optional, Dict
+import numpy as np  # type: ignore
+
+import tvm
+from tvm import TVMError
+from tvm.ir.module import IRModule
+from tvm.runtime.container import tuple_object
+from tvm.runtime.ndarray import NDArray
+
+from .setup_trainer import SetupTrainer
+from ..vm import VirtualMachine, build
+
+
+class Trainer:
+    r"""Unified wrapper for relax training. It uses SetupTrainer to setup the backbone and
+    then compiles/builds/runs the module. Also it maintains the parameters internally.
+
+    Parameters
+    ----------
+    backbone : IRModule
+        Backbone of the module to be trained. It should be a relax module with a function
+        which has name `predict` and returns a single Tensor.
+
+    params_num : int,
+        The numer of parameters. The first `params_num` inputs of the prediction function
+        will be regarded as the parameters. It will be annotated into the prediction function
+        as the value of func attr `update_params`.
+
+    setup_trainer_pass : SetupTrainer
+        The configured SetupTrainer pass.
+
+    Examples
+    --------
+    >>> setup_trainer = SetupTrainer()
+    >>> setup_trainer.set_loss(MSELoss(reduction="sum"), pred_sinfo, pred_sinfo)
+    >>> setup_trainer.set_optimizer(optim_type=SGD, lr=0.001)
+    >>> trainer = Trainer(MLP, 2, "main")
```

Comment:
Is the third parameter still a name now? I feel like it is the SetupTrainer object?
```suggestion
    >>> trainer = Trainer(MLP, 2, ___)
```